proposals/nnnn-video-streaming-state.md

```diff
@@ -0,0 +1,165 @@
+
+
+## Potential downsides
+On iOS devices, we still lack an effective way to bring a background Navigation/Projection app to foreground so that the app can restart/resume video streaming when the driver brings the app back to `FULL` in HMI.
+
+
+## Impact on existing code
+
+- RPC need get updated with new parameter. 
+- SDL core and mobile proxy lib need updates to support this new parameter. 
+- App's HMI level transition logic in SDL core need updates.
+
+
+## Alternatives considered
+
+- Instead of using a new enum data type `VideoStreamingState`, we can use just `Boolean` type and rename `videoStreamingState` to `isVideoSource` as the following.
+
+```xml
+<function name="OnHMIStatus" functionID="OnHMIStatusID" messagetype="notification">
+    ...
+    
+    <!-- new additions-->
+    <param name="isVideoSource" type="Boolean" mandatory="true">
+        <description>Indicates whehter an app can use video service or not.
+        If it is true, the mobile app can use video service.
+        If it is false, the mobile app must stop streaming video to SDL.        
+        </description>
+    </param>
+</function>
+```
+
+- Another alternative is that we can seperate one `STREAMABLE` state into two sub-states : `STREAMABLE_VISIBLE` and `STREAMABLE_NOT_VISIBLE`. However, this makes videoStreamingState coupled with HMI level. `VISIBLE` must be with `FULL` and `NOT_VISIBLE` must be with `LIMITED`.
+
+```xml
+<enum name="VideoStreamingState">
+    <description>Enumeration that describes possible states of video streaming. </description>
+    <element name="STREAMABLE_VISIBLE" />
+    <element name="STREAMABLE_NOT_VISIBLE" />
+    <element name="NOT_STREAMABLE" />
+</enum>
+```
+
+- With the navigation apps, paramter `audioStreamingState` was extened from the original design purpose (an indicator just for A2DP BT audio service) to include new binary audio service. Along the same line of this proposal, add a new paratmer `isAudioSource` for binary audio service only and make existing `audioStreamingState` just for A2DP aduio. This makes things clear but needs more modifications to the existing code.
+
+```xml
+<function name="OnHMIStatus" functionID="OnHMIStatusID" messagetype="notification">
+    ...
+    
+    <!-- new additions-->
+    <param name="isAudioSource" type="Boolean" mandatory="true">
+        <description>Indicates whehter an app can use binary audio service or not.
+        If it is true, the mobile app can use binary aduio service.
+        If it is false, the mobile app must stop streaming binary audio to SDL.        
+        </description>
+    </param>
+    <param name="isVideoSource" type="Boolean" mandatory="true">
+        <description>Indicates whehter an app can use video service or not.
+        If it is true, the mobile app can use video service.
+        If it is false, the mobile app must stop streaming video to SDL.        
```

Comment:
Should this be "...video to SDL Core." ?

Reply:
changed